src/main/java/net/dv8tion/jda/api/events/Event.java

```diff
@@ -71,4 +76,20 @@ public long getResponseNumber()
     {
         return responseNumber;
     }
+
+    @Nonnull
+    @Override
+    public DataObject getRawData()
+    {
+        if (rawData == null)
+            throw new IllegalStateException("This event does not contain any raw data, or event passthrough is not enabled, see JDABuilder#setEventPassthrough(boolean)");
+
+        return rawData;
```

Comment:
I'm not sure about this `hasRawData` and `getRawData` w/ exception pattern.

I think we should remove `hasRawData` and instead return `null` if there is no data.
We should only throw here if `isEventPassthrough` is false.

Reply:
If we throw on `isEventPassthrough` false, we have to provide an easy way to check for this too.

Reply:
Do we? They set it in the JDABuilder from the start. We throw an exception stating "if you want to use this feature, enable it in the builder".